### PR TITLE
feat(VEG-1285): refactor app example by using zendesk react garden components

### PR DESCRIPTION
**Pull request by cl8au**
@zendesk/vegemite 🤞 

## Description
- chore: upgrade node to v18 with v5 webpack
- feat: uplift existing code by using react garden design
- doc: update README to contains React component and other updates
- doc: remove IE support in browserslist
- ci: bump up the node version for GA
- ci: ignore specific lint issue for legacy code

Zendesk CSS components are out of support. We'd like to uplift this scaffold to rely on Zendesk React components.

## Tasks
- [x] Write tests

## References
- JIRA: https://zendesk.atlassian.net/browse/VEG-1285
- PR: This PR is very similar as https://github.com/zendesk/app_scaffold/pull/143 which contains almost identical code changes

## Screenshots (if needed)
Screen Shot 2022-12-22 at 16 58 57

Screen Shot 2022-12-22 at 16 59 59

## Risks
* [Medium]: Refactor the example app by using Zendesk React component

### CODE REVIEW

File: packages/react/spec/i18n.spec.js [hunk 1]
```diff
@@ -2,30 +2,38 @@
 import i18n from '../src/javascripts/lib/i18n'
 
 const mockEN = {
-  'one': 'the first translation',
+  one: 'the first translation',
```
Comment (Frederick888) [inline packages/react/spec/i18n.spec.js:5]: Did this trigger a lint warning or something?

**Review by Frederick888 — APPROVED** [review]
💯

File: packages/react/spec/i18n.spec.js (hunk 1 above)
Comment (cl8au) [reply packages/react/spec/i18n.spec.js:5]: oh.. no, I should revert it back in both PR :D nice pick, it does not change anything, just me for testing something before

Comment (cl8au) [reply packages/react/spec/i18n.spec.js]: Addressed  :)

Comment (cl8au) [reply packages/react/spec/i18n.spec.js:5]: Oh, it is lint fix actually. I will ignore this change then :)

**Review by JeanMarcGoepfert — APPROVED** [review]

**Pull request merged**